src/services/review/RepoCheckerService.ts

```diff
@@ -4,110 +4,358 @@ import path from "path"
-                ])
-              }
+        .andThen((setOfAllMediaAndPagesPath) => {
+          mdFiles.forEach(async (fileName) => {
+            const file = fs.readFileSync(path.join(repoPath, fileName), "utf8")
+
+            const filePermalink = file
+              .split("---")
+              ?.at(1)
+              ?.split("permalink: ")
+              ?.at(1)
+              ?.split("\n")
+              ?.at(0)
+
+            if (!filePermalink) {
+              return
             }
-          })
-          const imgTags = dom.window.document.querySelectorAll("img")
-          forEach(imgTags, (tag) => {
-            const src = tag.getAttribute("src")
-            if (src) {
-              const encodedSrc = this.documentPathDecoder(src)
-              if (
-                !this.isExternalLinkOrPageRef(encodedSrc) &&
-                !setOfAllMediaAndPagesPath.has(encodedSrc)
-              ) {
-                this.persistLogs(repo, [
-                  new SiteCheckerError(
-                    `Broken image ${src} found in ${fileName} which has permalink ${filePermalink}`
-                  ),
-                ])
-              }
+
+            // we are only parsing out the front matter
+            const fileContent = file.split("---")?.slice(2).join("---")
+            if (!fileContent) {
+              return
             }
+            const html = await marked.parse(fileContent)
+            const dom = new JSDOM(html)
+            const anchorTags = dom.window.document.querySelectorAll("a")
+            forEach(anchorTags, (tag) => {
+              const href = tag.getAttribute("href")
+              const text = tag.textContent || ""
+              if (href) {
+                const encodedHref = this.mediaPathDecoder(href)
+                if (
+                  !this.isExternalLinkOrPageRef(encodedHref) &&
+                  !setOfAllMediaAndPagesPath.has(encodedHref) &&
+                  // Amplify supports adding of trailing slash
+                  !setOfAllMediaAndPagesPath.has(`${encodedHref}/`)
+                ) {
+                  const error: RepoError = {
+                    type: RepoErrorTypes.BROKEN_LINK,
+                    linkToAsset: href,
+
+                    viewablePageInCms: this.getPathInCms(fileName),
+                    viewablePageInStaging: path.join(
+                      stagingLink,
+                      filePermalink
+                    ),
+                    linkedText: text,
+                  }
+                  errors.push(error)
+                }
+              }
+            })
+            const imgTags = dom.window.document.querySelectorAll("img")
+            forEach(imgTags, (tag) => {
+              const src = tag.getAttribute("src")
+              if (src) {
+                const encodedSrc = this.mediaPathDecoder(src)
+                if (
+                  !this.isExternalLinkOrPageRef(encodedSrc) &&
+                  !setOfAllMediaAndPagesPath.has(encodedSrc)
```

Comment:
this might potentially be buggy - our upstream has a list of [media](https://github.com/isomerpages/isomerpages-template/tree/staging/assets/img) that won't be present here, but will exist in the output. 

we might wanna add those to an excludes list or run this on the final built site, when all assets are pulled